src/ui/components/ide/monaco-editor.tsx

```diff
@@ -36,13 +37,29 @@ export default function (props: { onChange: OnChange }) {
 		ide: { refreshIDE, syncMonacoModels, getActiveFile },
 	} = useCommands();
 
+	// Add a state for theme (light/dark)
+	const savedTheme = (localStorage.getItem('editorTheme') as 'dark' | 'light') || 'dark';
```

Comment:
Can we try this rather than doing it manually? https://zustand.docs.pmnd.rs/integrations/persisting-store-data 

PS: We already use zustand.

Reply:
ok, so how about I create a `theme.store.ts` in `state-managers` and implement the logic to persist theme state using zustand ?

Reply:
Sounds good!